fs/shm/shmfs.c

```diff
@@ -391,13 +347,13 @@ static int shmfs_unmap(uintptr_t vaddr, size_t length)
   /* Add the virtual memory back to the shared memory pool */
 
   shm_free(group, (FAR void *)vaddr, length);
```

Comment:
This is still going to fail if the mappings that are being freed do not belong to the currently active address environment.

Reply:
Mmap still needs to be done in the driver, because the driver only knows what needs to be mapped. So maybe it is after all cleaner to also do also unmapping here? I guess that the group is not the current one only in some special case, like when deleting the whole address env? Should we perhaps just pass a boolean telling whether the mmu mapping should be dropped or not (bool cleanup_only or something like that?). Or pass NULL as group in such case?

Reply:
Or have it asymmetric in this regard, do unmappin in upper layer and mmap in driver?

Reply:
One option is, since shm_free only works when the group is current one, to just check that the group is current one inside shm_free, returning error if it is not?

Reply:
I don't have a good answer nor a clean / neat way of doing this. Still, here are some remarks:

- The only process that can map/unmap ANYTHING from MMU, is the current process / owner of the mapping
- The only case when mmap/munmap is called, is when the process itself wants to map/unmap something

- The problem we are solving, is what happens when a process is dying. There is NO NEED to unmap in this case, only release the inode references. 
- Only the process itself can kill itself, or if the kernel does OOM / kernel panic it gets murdered forcibly, in either case the MMU mappings will be removed and no unmap is needed here. Again, only the inode references need to be released.

Reply:
> Or have it asymmetric in this regard, do unmappin in upper layer and mmap in driver?

I do like the symmetry, either or.

Reply:
> One option is, since shm_free only works when the group is current one, to just check that the group is current one inside shm_free, returning error if it is not?

It can be tested there, but calling shm_free for a group from another group is an illegal operation.

Maybe one option like you said is to give group == NULL when calling munmap() to indicate that there is no group structure to whom the mapping belongs? IDK, this is a tough one...